### ISSUE TITLE
feature: followOn

### ISSUE THREAD

**Issue by Blahah** [ContentMine/scraperJSON#6]
```json
{
  "url": "\\w+",
  "name": "followOn example",
  "followable": {
    "figurePage": {
      "selector": "//a[@class='full-figure']",
      "attribute": "href"
    }
  },
  "elements" : {
    "figure": {
      "follow": "figurePage",
      "caption": { "selector": "//figcaption" },
      "img": {
        "selector": "//figure//img",
        "attribute": "src"
      }
   }
}
```

**Comment by blahah** [comment]
The idea is that `followables` can be created two ways:
1. any normal element can be followable
2. specific followable elements can be captured

The difference is that the elements under the `followable` key are not included in the results.

**Comment by blahah** [comment]
When we do the scraping, any normal element that is depended on by a follow will be prioritised.

**Comment by blahah** [comment]
This is implemented is Thresher v0.3 onwards.
